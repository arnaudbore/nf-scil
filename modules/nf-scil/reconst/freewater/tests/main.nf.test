nextflow_process {

    name "Test Process RECONST_FREEWATER"
    script "../main.nf"
    process "RECONST_FREEWATER"

    tag "modules"
    tag "modules_nfcore"
    tag "reconst"
    tag "reconst/freewater"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    config "./nextflow.config"

    setup {
            run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
                script "../../../../../subworkflows/nf-scil/load_test_data/main.nf"
                process {
                    """
                    input[0] = Channel.from( [ "commit_amico.zip" ] )
                    input[1] = "test.load-test-data"
                    """
                }
            }
        }

    test("reconst - freewater") {
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/dwi.nii.gz", checkIfExists: true),
                        file("\${test_data_directory}/dwi.bval", checkIfExists: true),
                        file("\${test_data_directory}/dwi.bvec", checkIfExists: true),
                        file("\${test_data_directory}/mask.nii.gz", checkIfExists: true),
                        []
                    ]
                 }
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("reconst - freewater_no_mask") {
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/dwi.nii.gz", checkIfExists: true),
                        file("\${test_data_directory}/dwi.bval", checkIfExists: true),
                        file("\${test_data_directory}/dwi.bvec", checkIfExists: true),
                        [],
                        []
                    ]
                 }
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
