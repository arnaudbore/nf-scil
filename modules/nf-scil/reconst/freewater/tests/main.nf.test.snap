{
    "reconst - freewater_no_mask": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    [
                        "A_001.npy:md5,11fe274731023f323146616ea5051c24",
                        "A_002.npy:md5,edc3da71ae9049a2bb768f3a9ce3b0d6",
                        "A_003.npy:md5,57324deb5eef1754ce5d5320bd4f44e2",
                        "A_004.npy:md5,074ff1605268308f029c0cfb1c9b2169",
                        "A_005.npy:md5,adca9ff0a85f38717fd6d771213d31d8",
                        "A_006.npy:md5,a59dafdf34e0135f20e22b185b017ff9",
                        "A_007.npy:md5,2d03444bae1b663f13b24cfb10b01bad",
                        "A_008.npy:md5,6a05008dbf09c6fb28481980a74bb356",
                        "A_009.npy:md5,1e083dab9c8253c4fb1ec1b60ef5d83f",
                        "A_010.npy:md5,3206fc1a74b43496d45c5e4c4bc30d5e",
                        "A_011.npy:md5,fdfafdc3d2d465a44988a54670d12c10"
                    ]
                ],
                "6": [
                    "versions.yml:md5,8c5d0f921a9f4a719aa37c7a45699960"
                ],
                "dir": [
                    
                ],
                "dwi_fw_corrected": [
                    
                ],
                "fibervolume": [
                    
                ],
                "fw": [
                    
                ],
                "kernels": [
                    [
                        "A_001.npy:md5,11fe274731023f323146616ea5051c24",
                        "A_002.npy:md5,edc3da71ae9049a2bb768f3a9ce3b0d6",
                        "A_003.npy:md5,57324deb5eef1754ce5d5320bd4f44e2",
                        "A_004.npy:md5,074ff1605268308f029c0cfb1c9b2169",
                        "A_005.npy:md5,adca9ff0a85f38717fd6d771213d31d8",
                        "A_006.npy:md5,a59dafdf34e0135f20e22b185b017ff9",
                        "A_007.npy:md5,2d03444bae1b663f13b24cfb10b01bad",
                        "A_008.npy:md5,6a05008dbf09c6fb28481980a74bb356",
                        "A_009.npy:md5,1e083dab9c8253c4fb1ec1b60ef5d83f",
                        "A_010.npy:md5,3206fc1a74b43496d45c5e4c4bc30d5e",
                        "A_011.npy:md5,fdfafdc3d2d465a44988a54670d12c10"
                    ]
                ],
                "nrmse": [
                    
                ],
                "versions": [
                    "versions.yml:md5,8c5d0f921a9f4a719aa37c7a45699960"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-05-22T17:34:55.041758"
    },
    "reconst - freewater": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    [
                        "A_001.npy:md5,11fe274731023f323146616ea5051c24",
                        "A_002.npy:md5,edc3da71ae9049a2bb768f3a9ce3b0d6",
                        "A_003.npy:md5,57324deb5eef1754ce5d5320bd4f44e2",
                        "A_004.npy:md5,074ff1605268308f029c0cfb1c9b2169",
                        "A_005.npy:md5,adca9ff0a85f38717fd6d771213d31d8",
                        "A_006.npy:md5,a59dafdf34e0135f20e22b185b017ff9",
                        "A_007.npy:md5,2d03444bae1b663f13b24cfb10b01bad",
                        "A_008.npy:md5,6a05008dbf09c6fb28481980a74bb356",
                        "A_009.npy:md5,1e083dab9c8253c4fb1ec1b60ef5d83f",
                        "A_010.npy:md5,3206fc1a74b43496d45c5e4c4bc30d5e",
                        "A_011.npy:md5,fdfafdc3d2d465a44988a54670d12c10"
                    ]
                ],
                "6": [
                    "versions.yml:md5,8c5d0f921a9f4a719aa37c7a45699960"
                ],
                "dir": [
                    
                ],
                "dwi_fw_corrected": [
                    
                ],
                "fibervolume": [
                    
                ],
                "fw": [
                    
                ],
                "kernels": [
                    [
                        "A_001.npy:md5,11fe274731023f323146616ea5051c24",
                        "A_002.npy:md5,edc3da71ae9049a2bb768f3a9ce3b0d6",
                        "A_003.npy:md5,57324deb5eef1754ce5d5320bd4f44e2",
                        "A_004.npy:md5,074ff1605268308f029c0cfb1c9b2169",
                        "A_005.npy:md5,adca9ff0a85f38717fd6d771213d31d8",
                        "A_006.npy:md5,a59dafdf34e0135f20e22b185b017ff9",
                        "A_007.npy:md5,2d03444bae1b663f13b24cfb10b01bad",
                        "A_008.npy:md5,6a05008dbf09c6fb28481980a74bb356",
                        "A_009.npy:md5,1e083dab9c8253c4fb1ec1b60ef5d83f",
                        "A_010.npy:md5,3206fc1a74b43496d45c5e4c4bc30d5e",
                        "A_011.npy:md5,fdfafdc3d2d465a44988a54670d12c10"
                    ]
                ],
                "nrmse": [
                    
                ],
                "versions": [
                    "versions.yml:md5,8c5d0f921a9f4a719aa37c7a45699960"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-05-22T17:34:44.701002"
    }
}